.github/actions/validate-deployment-settings/action.yml

```diff
@@ -0,0 +1,106 @@
+name: Validate Environment Deployment Settings
+description: Action that validates various `ACCESS-NRI/build-cd` deployment settings
+inputs:
+  settings-path:
+    type: string
+    required: false
+    default: './cd/config/settings.json'
+    description: Path to the build-cd settings.json file to validate
+  target:
+    type: string
+    required: false
+    default: Gadi
```

Comment:
Indeed it would! Something like:

```yml
verify-settings:
  runs-on: ubuntu-latest
  strategy:
    matrix: 
      target: ["Gadi". "Setonix"]  # this would more likely come from an earlier step that reads `.deployment | keys` in `config.settings.json` but still...
  steps:
    - uses: actions/checkout@v4
    - uses: access-nri/build-cd/.github/actions/validate-deployment-settings@main
      with:
        target: ${{ matrix.target }}
```